library/ssl_tls.c

```diff
@@ -1958,6 +1973,12 @@ static int ssl_tls13_session_save( const mbedtls_ssl_session *session,
 #if defined(MBEDTLS_SSL_CLI_C)
     if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
     {
+        size_t hostname_len = session->hostname == NULL?0:strlen( session->hostname );
```

Comment:
```suggestion
        size_t hostname_len = ( session->hostname == NULL ) ?
                              0 : strlen( session->hostname );
```
and declare `hostname_len` as a function local variable as it is used in different blocks of the function.